src/zulip.rs

```diff
@@ -203,6 +213,105 @@ fn handle_command<'a>(
     })
 }
 
+#[derive(PartialEq)]
+enum CommentType {
+    AssignIssuePriority,
+}
+
+// https://docs.zulip.com/api/outgoing-webhooks#outgoing-webhook-format
+#[derive(serde::Deserialize, Debug)]
+struct ZulipReply {
+    message: ZulipMessage,
+}
+
+#[derive(serde::Deserialize, Debug)]
+struct ZulipMessage {
+    subject: String, // ex.: "[weekly] 2023-04-13"
+}
+
+async fn get_zulip_msg(ctx: &Context, id: u64) -> anyhow::Result<ZulipReply> {
+    let bot_api_token = env::var("ZULIP_API_TOKEN").expect("ZULIP_API_TOKEN");
+    let zulip_resp = ctx
+        .github
+        .raw()
+        .get(format!(
+            "https://rust-lang.zulipchat.com/api/v1/messages/{}",
+            id
+        ))
+        .basic_auth(BOT_EMAIL, Some(&bot_api_token))
+        .send()
+        .await?;
+
+    let zulip_msg_data = zulip_resp.json::<ZulipReply>().await.expect("TODO");
+    log::debug!("Zulip reply {:?}", zulip_msg_data);
+    Ok(zulip_msg_data)
+}
+
+// Add a comment to a Github issue/pr issuing a @rustbot command
+async fn add_comment_to_issue(
+    ctx: &Context,
+    message: &Message,
+    mut words: impl Iterator<Item = &str> + std::fmt::Debug,
+    ty: CommentType,
+) -> anyhow::Result<String> {
+    // retrieve the original Zulip message to build the complete URL
+    let zulip_msg = get_zulip_msg(ctx, message.id).await?;
```

Comment:
here I get again the full message froim Zulip to rebuild the full message URL to backlink to Zulip che discussion about the assigned priority to the issue